hera_qm/utils.py

```diff
@@ -251,24 +272,48 @@ def metrics2mc(filename, ftype):
             d['ant_metrics'][metric].append([ant, pol, 1.])
 
     elif ftype == 'omnical':
-        from pyuvdata import UVCal
-        uvcal = UVCal()
-        uvcal.read_calfits(filename)
-        pol_dict = {-5: 'x', -6: 'y'}
-        d['ant_metrics']['omnical_quality'] = []
-        for pi, pol in enumerate(uvcal.jones_array):
-            try:
-                pol = pol_dict[pol]
-            except KeyError:
-                raise ValueError('Invalid polarization for ant_metrics in M&C.')
-            for ai, ant in enumerate(uvcal.ant_array):
-                val = np.median(uvcal.quality_array[ai, 0, :, :, pi], axis=0)
-                val = np.mean(val)
-                d['ant_metrics']['omnical_quality'].append([ant, pol, val])
-        if uvcal.total_quality_array is not None:
-            d['array_metrics']['omnical_total_quality'] = np.mean(uvcal.total_quality_array)
+        from hera_qm.omnical_metrics import load_omnical_metrics
+        full_mets = load_omnical_metrics(filename)
+        pols = full_mets.keys()
+
+        # iterate over polarizations (e.g. XX, YY, XY, YX)
+        for i, pol in enumerate(pols):
+            # unpack metrics from full_mets
+            metrics = full_mets[pol]
+
+            # pack array metrics
+            cat = 'omnical_metrics_'
+            for met in ['chisq_tot_avg', 'chisq_good_sol', 'ant_phs_std_max', 'ant_phs_std_good_sol']:
+                catmet = cat + met + '_{}'.format(pol)
+                try:
+                    if metrics[met] is None:
+                        continue
+                    d['array_metrics'][catmet] = metrics[met]
+                except:
+                    pass
+
+            # pack antenna metrics, only uses auto pol (i.e. XX or YY)
+            if pol not in ['XX', 'YY']:
+                continue
+            cat = 'omnical_metrics_'
+            for met in ['chisq_ant_avg', 'chisq_ant_std', 'ant_phs_std']:
+                catmet = cat + met
+                try:
+                    if metrics[met] is None:
+                        continue
+                    d['ant_metrics'][catmet] = [[a, metrics['ant_pol'].lower(), metrics[met][a]] for a in metrics[met]]
```

Comment:
This overwrites the previous polarization. For example, when I ran this with an omnical file with xx and yy, the yy iteration of this loop overwrite the xx iteration, and the output dictionary only had yy. There needs to be a check if `d['ant_metrics'][catmet]` already exists to append, rather than overwrite.

Reply:
ah yes, that was dumb.. thanks